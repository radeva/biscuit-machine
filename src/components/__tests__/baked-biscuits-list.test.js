import React from 'react';
import BakedBiscuitsList from '../baked-biscuits-list';
import { render, cleanup } from '@testing-library/react';

afterEach(cleanup);

describe('Baked Biscuits List', () => {
    it('should not show biscuits when count is zero', () => {
        const { getByTestId } = render(
            <BakedBiscuitsList
              bakedBiscuitsCount={0}
            />
          );
      
          const bakedBiscuitsList = getByTestId('biscuits-list');;
          expect(bakedBiscuitsList.children.length).toBe(0);
    });

    it('should show correct amount of biscuits', () => {
        const biscuitsCount = 5;
        const { getByTestId } = render(
            <BakedBiscuitsList
              bakedBiscuitsCount={biscuitsCount}
            />
          );
      
          const bakedBiscuitsList = getByTestId('biscuits-list');;
          expect(bakedBiscuitsList.children.length).toBe(biscuitsCount);
    });

    it('should be rendered correctly', () => {
        const biscuitsCount = 5;
        const { container } = render(<BakedBiscuitsList bakedBiscuitsCount={biscuitsCount} />);
        expect(container).toMatchSnapshot();
    });
});