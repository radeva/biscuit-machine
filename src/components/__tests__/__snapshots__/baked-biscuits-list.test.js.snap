// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Baked Biscuits List should be rendered correctly 1`] = `
<div>
  <ul
    class="biscuits-list"
    data-testid="biscuits-list"
  >
    <li>
      <svg>
        baked-biscuit.svg
      </svg>
    </li>
    <li>
      <svg>
        baked-biscuit.svg
      </svg>
    </li>
    <li>
      <svg>
        baked-biscuit.svg
      </svg>
    </li>
    <li>
      <svg>
        baked-biscuit.svg
      </svg>
    </li>
    <li>
      <svg>
        baked-biscuit.svg
      </svg>
    </li>
  </ul>
</div>
`;
